File: BOJ/10845_큐/10845_큐.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <queue>
using namespace std;

queue<int> q;

void push(int x)
{
  q.push(x);
}

int pop()
{
  if (q.empty())
  {
    return -1;
  }
  int x = q.front();
  q.pop();
  return x;
}

int size()
{
  return q.size();
}

int empty()
{
  if (q.empty())
  {
    return 1;
  }
  else
  {
    return 0;
  }
}

int front()
{
  if (q.empty())
  {
    return -1;
  }
  else
  {
    return q.front();
  }
}

int back()
{
  if (q.empty())
  {
    return -1;
  }
  else
  {
    return q.back();
  }
}

int main(int argc, char *argv[])
{
  int n;
  int num;
  string s;

  cin >> n;

  while (n--)
  {
    cin >> s;

    if (s == "push")
    {
      cin >> num;
      push(num);
    }
    else if (s == "pop")
    {
      cout << pop() << endl;
    }
    else if (s == "size")
    {
      cout << size() << endl;
    }
    else if (s == "empty")
    {
      cout << empty() << endl;
    }
    else if (s == "front")
    {
      cout << front() << endl;
    }
    else if (s == "back")
    {
      cout << back() << endl;
    }
  }

  return 0;
}

File: BOJ/10845_큐/10845_큐.java
import java.util.*;
import java.io.*;

public class Main{
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    Queue<Integer> queue = new LinkedList<>();
    
    int n = Integer.parseInt(br.readLine());
    int num;
    int back = -1;
    for(int i=0; i<n; ++i){
      StringTokenizer st = new StringTokenizer(br.readLine());
      String s = st.nextToken();
      if("push".equals(s)){
          num = Integer.parseInt(st.nextToken());
          back = num;
          queue.add(num);
      } else if("pop".equals(s)){
          if(queue.size() != 0) {
            System.out.println(queue.poll());
          } else {
            back = -1;
            System.out.println(-1);
          }
      } else if("size".equals(s)){
          System.out.println(queue.size());
      } else if("empty".equals(s)){
          if(queue.size() == 0){
              System.out.println(1);    
          } else {
              System.out.println(0);
          }
      } else if("front".equals(s)){
        if(queue.size() != 0) {
          System.out.println(queue.peek());
        } else {
          System.out.println(-1);
        }
      } else if("back".equals(s)){
        if(queue.size() != 0) {
          System.out.println(back);
        } else {
          System.out.println(-1);
        }
      }
    }
  }
}